l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -211,6 +213,46 @@ def button_generate(self):
         finally:
             # tmpdir also contains uncompressed auditfile.
             shutil.rmtree(tmpdir)
+        self._release_in_use()
+
+    def _acquire_in_use(self):
+        """Lock record for this by current thread of process.
+
+        Use separate cursor, to force immediate visibility of update.
+        """
+        self.ensure_one()
+        lock_file_path = self._get_lockfile_path()
+        if os.path.exists(lock_file_path):
+            raise exceptions.UserError(
+                _("Generation of auditfile is already in progress.")
+            )
+        lock_info = "User {user} already generating auditfile {auditfile}".format(
+            user=self.env.user.display_name,
+            auditfile=self.name,
+        )
+        with open(lock_file_path, "w+") as tmphandle:
+            tmphandle.write(lock_info)
```

Comment:
Unfortunately there are Odoo instances around that have multiple servers connecting to the same database, so if you implement a locking mechanism, it's better to have it database-based than file-based.

Reply:
Ok, you're using the filestore folder; clever. Well, let's see what other reviewers say :-)